scripts/ci/build.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+((${BASH_VERSION%%.*} >= 4)) || { echo >&2 "$0: Error: Please upgrade Bash."; exit 1; }
+
+set -euxo pipefail
+
+if [[ "${OSTYPE}" == msys* ]]; then # Windows
+
+    cpus="${NUMBER_OF_PROCESSORS}"
+
+elif [[ "${OSTYPE}" == darwin* ]]; then # macOS
+
+    cpus="$(sysctl -n hw.ncpu)"
```

Comment:
`nproc` doesn't appear to be installed on macOS by default. At least not on my mac. It might be on GitHub Actions I suppose.

The `-j` option does no harm with `XCode` and `MSVC` and is needed if somebody wants to build with other generators on Windows or macOS.

Reply:
But we are not going to do so on CI :) If we want users to reuse the scripts - we should remove `brew` and use pwsh on windows.

Reply:
And yes, turns out you are right - nproc is a part of coreutils.

Reply:
If Windows users have Git installed then they have Git Bash. I think we should force them to use it for the sake of cross-platform compatibility (the CMake files will work without it of course).

It is even possible to use Git Bash from within [Windows Terminal](https://github.com/Microsoft/Terminal#installing-and-running-windows-terminal) by adding the following to settings.json:

```jsonc
// Windows Terminal's settings.json
// Add this to the "profiles" section to enable Git Bash. Adapt the
// "commandline" and "icon" paths depending on where you installed Git.
        {
            "guid": "{00000000-0000-0000-0000-000000000001}", // or make up your own GUID
            "name": "Bash",
            "commandline": "\"%UserProfile%\\scoop\\apps\\git\\current\\usr\\bin\\bash.exe\" -i -l",
            "icon": "%UserProfile%\\scoop\\apps\\git\\current\\usr\\share\\git\\git.ico",
            "startingDirectory": "%USERPROFILE%"
        }, // <-- Pay attention to commas in JSON files. They are needed between "}" and "{", but never before "}".
```

Reply:
There are multiple ways of installing Git on Windows :) 

> I think we should force them to use

I don't use git-bash and see no reason why I should :) Also, lately, I spent two days trying to get codesigning work, and it turned out that SignTool is simply incompatible with git-bash. So now we have a step that uses `pwsh`. (MuseScore uses regular batch file as well)